main.c

```diff
@@ -788,6 +861,7 @@ static int dev_added(struct tcmu_device *dev)
 	if (ret < 0)
 		goto cleanup_io_work_queue;
 
+	rhandler->dev = dev;
```

Comment:
I don't think you get a different handler instance per device so this overwrites a previous device.

If it's not possible to do a callback per device, then I think for on_mediachange you need to pass in the device name string in the format used in the netlink functions so we can just look up the device and use it. We could put it at the front of the cfgstring string.

I'm guessing it's possible but it would get ugly for with 1000s of devices it would be too many resources, so I guess just passing in the device in the cfg string is ok.

Reply:
So before, I fix this part, do you want a specific way of exposing tcmu_device to the mediachange function? As of now tcmur_handler has no way of getting to a tcmu_device which would then allow a search for the device via the name passed from targetcli. - One way could possibly be to add a darray into tcmur_handler that would contain a list of the tcmu_device pointers and then search via that...